packages/web_ui/src/components/SlavesPage.jsx

```diff
@@ -50,30 +66,105 @@ function GenerateSlaveTokenButton(props) {
+				<div className="codeblock">
+					<CopyButton
+						message={"Copied auth token to clipboard"}
+						text={token}
+					/>
+					{token}
+				</div>
+				<Typography.Paragraph>
+					You can set the token on an existing slave with the following command:
+				</Typography.Paragraph>
+				<div className="codeblock">
+					<CopyButton
+						message={"Copied configuration commands to clipboard"}
+						text={`npx clusterioslave config set slave.master_token ${token}
+							   npx clusterioslave config set slave.id ${slaveId}`}
+					/>
+					<p>npx clusterioslave config set slave.master_token &lt;token&gt;</p>
+					<p>npx clusterioslave config set slave.id &lt;slaveId&gt;</p>
+				</div>
+				<Typography.Paragraph>
+					Example slave setup commands:
+				</Typography.Paragraph>
+				<div className="codeblock">
+					<CopyButton
+						message={"Copied slave setup commands to clipboard"}
+						text={`
+							mkdir clusterio
+							cd clusterio
+							`+
+							// eslint-disable-next-line max-len
+							`npm init "@clusterio" -- --master-token ${token} --mode "slave" --download-headless --master-url ${document.location.origin}/ --slave-name "Slave ${slaveId || "?"}" --public-address localhost ${pluginString.length ? "--plugins" : ""} ${pluginString}`
+						}/>
+					<p>&gt; mkdir clusterio</p>
+					<p>&gt; cd clusterio</p>
+					<p>
+						&gt; npm init "@clusterio" --
+						--master-token <span className="highlight">{token} </span>
+						--mode "slave"
+						--download-headless
+						--master-url <span className="highlight">{document.location.origin}/ </span>
+						--slave-name <span className="highlight">"Slave {slaveId || "?"}" </span>
+						--public-address <span className="highlight">localhost </span>
+						{pluginString.length ? "--plugins" : ""} <span className="highlight">{pluginString}</span>
+					</p>
+					<p>&gt; ./run-slave</p>
+				</div>
+			</>}
 		</Modal>
 	</>;
 }
 
+function CopyButton({ text, message }) {
+	let [clipboardPermision, setClipboardPermission] = useState("granted");
+	useEffect(() => {
+		(async () => {
+			let result = await navigator.permissions.query({ name: "clipboard-write" });
+			// result.state is "granted", "denied" or "prompt"
+			setClipboardPermission(result.state);
+			result.onchange = function () {
+				setClipboardPermission(result.state);
+			};
+		})();
+	}, []);
+
+	async function checkClipboardPermission() {
+		let result = await navigator.permissions.query({ name: "clipboard-write" });
+		// result.state is "granted", "denied" or "prompt"
+		setClipboardPermission(result.state);
+		result.onchange = function () {
+			setClipboardPermission(result.state);
+		};
+		return result.state === "granted";
+	}
+
+	return <Button
+		className="copy-button"
+		danger={clipboardPermision !== "granted"}
+		onClick={async () => {
+			if (await checkClipboardPermission()) {
+				navigator.clipboard.writeText(text);
+				notify(message);
+			}
+		}}
+	>
+		<CopyOutlined />
+	</Button>;
+}
```

Comment:
This new copy button just shows up as red in my browser and doesn't do anything when I click on it?

![image](https://user-images.githubusercontent.com/14362102/179361705-c23ecfa0-eba2-401d-bb87-4ac00c3be1b3.png)

Also the spacing of the text is wrong, there should be more space from the code block to the unrelated text below it than there is from the code block to the related text above it.